src/MLJXGBoostInterface.jl

```diff
@@ -1,770 +1,212 @@
-    top_k::Int
-    tweedie_variance_power::Float64
-    objective
-    base_score::Float64
-    eval_metric
-    seed::Int
-    nthread::Int
+function importances(X, r)
+    fs = schema(X).names
+    [named_importance(fi, fs) for fi ∈ XGB.importance(r)]
 end
 
-
-"""
-    XGBoostCount(; seed=0, kwargs...)
-
-The XGBoost model for targets with `Count` scitype. Gives
-deterministic (point) predictions. For admissible `kwargs`, see
-[https://xgboost.readthedocs.io/en/latest/parameter.html](https://xgboost.readthedocs.io/en/latest/parameter.html).
-
-For a time-dependent random seed, use `seed=-1`.
-
-See also: XGBoostRegressor, XGBoostClassifier
-
-"""
-function XGBoostCount(
-    ;num_round=100
-    ,booster="gbtree"
-    ,disable_default_eval_metric=0
-    ,eta=0.3
-    ,gamma=0
-    ,max_depth=6
-    ,min_child_weight=1
-    ,max_delta_step=0
-    ,subsample=1
-    ,colsample_bytree=1
-    ,colsample_bylevel=1
-    ,lambda=1
-    ,alpha=0
-    ,tree_method="auto"
-    ,sketch_eps=0.03
-    ,scale_pos_weight=1
-    ,updater="auto"
-    ,refresh_leaf=1
-    ,process_type="default"
-    ,grow_policy="depthwise"
-    ,max_leaves=0
-    ,max_bin=256
-    ,predictor="cpu_predictor" #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type="uniform"
-    ,normalize_type="tree"
-    ,rate_drop=0.0
-    ,one_drop=0
-    ,skip_drop=0.0
-    ,feature_selector="cyclic"
-    ,top_k=0
-    ,tweedie_variance_power=1.5
-    ,objective="count:poisson"
-    ,base_score=0.5
-    ,eval_metric="rmse"
-    ,seed=0
-    ,nthread = 1)
-
-    model = XGBoostCount(
-    num_round
-    ,booster
-    ,disable_default_eval_metric
-    ,eta
-    ,gamma
-    ,max_depth
-    ,min_child_weight
-    ,max_delta_step
-    ,subsample
-    ,colsample_bytree
-    ,colsample_bylevel
-    ,lambda
-    ,alpha
-    ,tree_method
-    ,sketch_eps
-    ,scale_pos_weight
-    ,updater
-    ,refresh_leaf
-    ,process_type
-    ,grow_policy
-    ,max_leaves
-    ,max_bin
-    ,predictor #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type
-    ,normalize_type
-    ,rate_drop
-    ,one_drop
-    ,skip_drop
-    ,feature_selector
-    ,top_k
-    ,tweedie_variance_power
-    ,objective
-    ,base_score
-    ,eval_metric
-    ,seed
-    ,nthread)
-
-     message = MMI.clean!(model)
-     isempty(message) || @warn message
-
-    return model
-end
-
-function MMI.clean!(model::XGBoostCount)
-    warning = ""
-    if(!(model.objective in ["count:poisson"]))
-        warning *= "Changing objective to \"poisson\", "*
-                       "the only supported value. "
-        model.objective="poisson"
+function MMI.feature_importances(model::XGTypes, (booster, _), (features,))
+    dict = XGB.importance(booster, model.importance_type)
+    if length(last(first(importance_dict))) > 1
+        [features[k] => zero(first(v)) for (k, v) in dict]
+    else
+        [features[k] => first(v) for (k, v) in dict]
```

Comment:
`importance_dict` does not seem to be defined. I suppose this should be `dict`? 

Is there a test for this functionality?

Reply:
Yeah, there were a number of things wrong here, I have added a unit test.